packages/bpk-component-navigation-stack/src/BpkNavigationStack.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import { cssModules } from 'bpk-react-utils';
+import React, { Children, type Element } from 'react';
+import PropTypes from 'prop-types';
+import { durationSm } from 'bpk-tokens/tokens/base.es6';
+import { TransitionGroup, Transition } from 'react-transition-group';
+import debounce from 'lodash/debounce';
+
+import { isRTL } from './utils';
+import STYLES from './BpkNavigationStack.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Views = Array<Element<any>>;
+
+export type Props = {
+  views: Views,
+  className: ?string,
+};
+
+type Transitions = 'entering' | 'leaving';
+
+type State = {
+  transition: ?Transitions,
+};
+
+class BpkNavigationStack extends React.Component<Props, State> {
+  static propTypes = {
+    views: PropTypes.arrayOf(PropTypes.element).isRequired,
+    className: PropTypes.string,
+  };
+
+  static defaultProps = {
+    className: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      transition: undefined,
```

Comment:
`null` would be better than `undefined` right? `null` explicitly means no value whereas `undefined` is up to interpretation...

Reply:
yeah, I was having problems with flow and put undefined just to make it happy, but forgot to change